lib/faraday/raise_errors.rb

```diff
@@ -8,10 +8,10 @@ class RaiseErrors < ::Faraday::Response::Middleware
     def on_complete(env)
       case env[:status]
       when 404
-        raise Faraday::Error::ResourceNotFound, response_values(env)
+        raise Faraday::ResourceNotFound, response_values(env)
       when 407
         # mimic the behavior that we get with proxy requests with HTTPS
-        raise Faraday::Error::ConnectionFailed, %{407 "Proxy Authentication Required "}
+        raise Faraday::ConnectionFailed, %{407 "Proxy Authentication Required "}
```

Comment:
You know, these lines were originally copied from [here](https://github.com/lostisland/faraday/blob/master/lib/faraday/response/raise_error.rb#L21-L26), which seems to have expanded.

Another way of handling this — that might work for both — might just be to `return` from this middleware on 404 and 407, which I think would fall through to Faraday's middleware.

... Is `Faraday::Response::RaiseError` included by default?

Reply:
It looks like it's not. It took some digging, but I think it's only `request :url_encoding` that's included by default: https://github.com/lostisland/faraday/blob/f6401225a4c3d778997b14c4be08a3f56aa414fe/lib/faraday/rack_builder.rb#L66-L68

The fact that `Faraday::Response::RaiseError` has expanded as it has, it might be a good reason to eventually refactor away from using this middleware. Is the main benefit we get from this middleware the grouping of errors into Client/Server-specific subclasses? 🤔